docs/openapi.yml

```diff
@@ -1344,6 +1344,45 @@ components:
     ProjectType:
       description: the build type for the project. Built-in types are 'liberty', 'spring', 'swift', 'nodejs', and 'docker'.
       type: string
+    ProjectTypeDescriptor:
+      description: Descriptor of a project type, including possible subtypes
+      type: object
+      required:
+        - projectType
+        - projectSubtypes
+      properties:
+        projectType:
+          $ref: '#/components/schemas/ProjectType'
+        projectSubtypes:
+          $ref: '#/components/schemas/ProjectSubtypesDescriptor'
+    ProjectSubtype:
+      description: A project subtype defined by a project extension
+      type: object
+      required:
+        - id
+        - label
+      properties:
+        id:
+          type: string
+        version:
+          type: string
+        label:
+          type: string
+        description:
+          type: string
+    ProjectSubtypesDescriptor:
+      description: Descriptor for subtypes of a project type
+      type: object
+      required:
+        - items
+      properties:
+        prompt:
+          description: A prompt for user to select a subtype
```

Comment:
We are not sure how else will the name of the subtype will need to be exposed on the UI in the future.  Instead of having a dedicated question for subtype selection prompt, should this be only the label of the subtype, e.g. Appsody stack, instead of a full question?  The problem with the full question prompt is that we need to come up with different strings in case we need to prompt users on subtype related things or error messages, we need to come up with unique strings for each one. We may be better off just provide the label of the subtype and let the IDE decide the full prompt by doing string substitution with the subtype lable only.

Reply:
Sure, can do that.  And should we simply rename this to `label`?

Reply:
@elsony the change from prompt to label has been done. We have also tested this via the appsody extension via the merge of https://github.com/eclipse/codewind-appsody-extension/pull/32. Is there anything else required for a review approval? Thank you